build/source/driver/multi_driver.f90

```diff
@@ -247,77 +357,123 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
- call v_shortcut(err,message); call handle_err(err,message) ! calculate "short-cut" variables such as volumetric heat capacity
+
+ ! calculate vertical distribution of root density
+ call rootDensty(mparStruct%hru(iHRU)%var,& ! vector of model parameters
+                 indxStruct%hru(iHRU),    & ! data structure of model indices
+                 progStruct%hru(iHRU),    & ! data structure of model prognostic (state) variables
+                 diagStruct%hru(iHRU),    & ! data structure of model diagnostic variables
+                 err,message)               ! error control
+ call handle_err(err,message) 
+
+ ! calculate saturated hydraulic conductivity in each soil layer
+ call satHydCond(mparStruct%hru(iHRU)%var,& ! vector of model parameters
+                 indxStruct%hru(iHRU),    & ! data structure of model indices
+                 progStruct%hru(iHRU),    & ! data structure of model prognostic (state) variables
+                 fluxStruct%hru(iHRU),    & ! data structure of model fluxes 
+                 err,message)               ! error control
+ call handle_err(err,message)
+
+ ! calculate "short-cut" variables such as volumetric heat capacity
+ call v_shortcut(mparStruct%hru(iHRU)%var,& ! vector of model parameters
+                 diagStruct%hru(iHRU),    & ! data structure of model diagnostic variables
+                 err,message)               ! error control
+ call handle_err(err,message)
+
  ! overwrite the vegetation height
- HVT(type_data%var(iLookTYPE%vegTypeIndex)) = mpar_data%var(iLookPARAM%heightCanopyTop)
- HVB(type_data%var(iLookTYPE%vegTypeIndex)) = mpar_data%var(iLookPARAM%heightCanopyBottom)
+ HVT(typeStruct%hru(iHRU)%var(iLookTYPE%vegTypeIndex)) = mparStruct%hru(iHRU)%var(iLookPARAM%heightCanopyTop)
+ HVB(typeStruct%hru(iHRU)%var(iLookTYPE%vegTypeIndex)) = mparStruct%hru(iHRU)%var(iLookPARAM%heightCanopyBottom)
+
  ! overwrite the tables for LAI and SAI
  if(model_decisions(iLookDECISIONS%LAI_method)%iDecision == specified)then
-  SAIM(type_data%var(iLookTYPE%vegTypeIndex),:) = mpar_data%var(iLookPARAM%winterSAI)
-  LAIM(type_data%var(iLookTYPE%vegTypeIndex),:) = mpar_data%var(iLookPARAM%summerLAI)*greenVegFrac_monthly
+  SAIM(typeStruct%hru(iHRU)%var(iLookTYPE%vegTypeIndex),:) = mparStruct%hru(iHRU)%var(iLookPARAM%winterSAI)
+  LAIM(typeStruct%hru(iHRU)%var(iLookTYPE%vegTypeIndex),:) = mparStruct%hru(iHRU)%var(iLookPARAM%summerLAI)*greenVegFrac_monthly
  endif
+
  ! initialize canopy drip
  ! NOTE: canopy drip from the previous time step is used to compute throughfall for the current time step
- mvar_hru(iHRU)%var(iLookMVAR%scalarCanopyLiqDrainage)%dat(1) = 0._dp  ! not used
- ! define the filename for model spinup
- write(fileout,'(a,i0,a,i0,a)') trim(OUTPUT_PATH)//trim(OUTPUT_PREFIX)//'_spinup'//trim(output_fileSuffix)//'.nc'
- ! define the file if the first parameter set
+ fluxStruct%hru(iHRU)%var(iLookFLUX%scalarCanopyLiqDrainage)%dat(1) = 0._dp  ! not used
```

Comment:
This variable actually is used (the comment is incorrect). The variable `scalarCanopyLiqDrainage` needs to be initialized here, since the unloading flux in `canopySnow` is computed using canopy drip (scalarCanopyLiqDrainage) from the previous time step.